src/rgw/rgw_http_client.cc

```diff
@@ -1167,6 +1167,7 @@ void *RGWHTTPManager::reqs_thread_entry()
 	curl_easy_getinfo(e, CURLINFO_RESPONSE_CODE, (void **)&http_status);
 
 	int status = rgw_http_error_to_errno(http_status);
+	req_data->client->set_http_status(http_status);
```

Comment:
i don't think it's safe to modify the client here outside of its mutex; instead, i think we need to feed this `http_status` through `RGWHTTPManager::finish_request()` -> `rgw_http_req_data::finish()`, so that finish() can call `client->set_http_status()` under its lock

Reply:
Thank you for your careful review, I have made the modification, please help to review again.

Reply:
can you please instead add http_status as an argument to finish_request(), so it can pass that on to rgw_http_req_data::finish()? that way it gets the http_status without having to take the lock twice